typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -141,12 +172,38 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
         await warpRoute[method]();
         if (tokenType === TokenType.collateral) {
           const wrappedToken = await warpRoute.wrappedToken();
-          const xerc20 = IXERC20__factory.connect(wrappedToken, this.provider);
           try {
+            const xerc20 = IXERC20__factory.connect(
+              wrappedToken,
+              this.provider,
+            );
             await xerc20['mintingCurrentLimitOf(address)'](warpRouteAddress);
             return TokenType.XERC20;
-            // eslint-disable-next-line no-empty
-          } catch {}
+          } catch (error) {
+            this.logger.debug(
+              `Warp route token at address "${warpRouteAddress}" on chain "${this.chain}" is not a ${TokenType.XERC20}`,
+              error,
+            );
+          }
+
+          try {
```

Comment:
why do we need separate try-catch? aren't these 2 mutually exclusive? in this case, we are inferring to the reader that the warp route can be xerc20 and fiat token.

perhaps we should make use of `continue`? or perhaps another way?

Reply:
It can't becuase if we find out that the token is an XERC20 then we just return inside the try catch, same for the fiat collateral :
- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/e2c7ba45fa7f45db2e00baddf999ff5f9a544b70/typescript/sdk/src/token/EvmERC20WarpRouteReader.ts#L175-L186
- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/e2c7ba45fa7f45db2e00baddf999ff5f9a544b70/typescript/sdk/src/token/EvmERC20WarpRouteReader.ts#L189-L200

Reply:
ah yes my mistake. misread this